scripts/configure-cluster/common/setup-operators.sh

```diff
@@ -31,13 +34,40 @@ install_service_binding_operator(){
 EOF
 }
 
+deploy_service_binding_operator_master(){
+  oc create -f - <<EOF
+  apiVersion: operators.coreos.com/v1alpha1
+  kind: CatalogSource
+  metadata:
+    name: service-binding-master
+    namespace: openshift-marketplace
+  spec:
+    displayName: Service Binding Operator build from master
+    image: quay.io/redhat-developer/servicebinding-operator:index
+    priority: 500
+    publisher: Red Hat
+    sourceType: grpc
+    updateStrategy:
+      registryPoll:
+        interval: 10m0s
+EOF
+}
+
 if [ $KUBERNETES == "true" ]; then
   # install "redis-oprator" using "kubectl" in "operators" namespace; use "operatorhubio-catalog" catalog source from "olm" namespace
   install_redis_operator kubectl operators operatorhubio-catalog olm
 else
+  if [$NIGHTLY == "true"]; then
+    # Deploy SBO master catalog source on OCP Nightly test run
+    deploy_service_binding_operator_master
+
+    SBO_SOURCE="service-binding-master"
+    SBO_NAME="service-binding-operator"
```

Comment:
What's the point of setting these variables after installing the Operator? I'm probably missing something here.

Reply:
Ohh under if statement I am just trying to deploy `SBO` master and set the env var. Installation of operator steps are common for `Nightly` and `Openshift`, so to avoid duplication I kept those steps out of if block which will run for both `nightly` and `openshift` regular pr. I hope it clears your doubt 🙂

Reply:
> I kept those steps out of if block which will run for both nightly and openshift regular pr.

But it's actually inside the `if` block. It will get executed only if the check `if [ $NIGHTLY == true ]` is satisfied. Or am I still missing something?

Reply:
> > I kept those steps out of if block which will run for both nightly and openshift regular pr.
> 
> But it's actually inside the `if` block. It will get executed only if the check `if [ $NIGHTLY == true ]` is satisfied. Or am I still missing something?

No its not under if block https://github.com/openshift/odo/pull/4958/files#diff-510a35137f161ddca098f30a15bcaac95c9f4f33b75a0bd861a7d8e77a56b1f1R72

Executed steps:
```
$ ./scripts/configure-cluster/common/setup-operators.sh 
+ export SBO_CATALOG_SOURCE=redhat-operators
+ SBO_CATALOG_SOURCE=redhat-operators
+ export SBO_SUBSCRIPTION_NAME=rh-service-binding-operator
+ SBO_SUBSCRIPTION_NAME=rh-service-binding-operator
+ '[' == true ']'
./scripts/configure-cluster/common/setup-operators.sh: line 56: [: ==: unary operator expected
+ '[' true == true ']'
+ deploy_service_binding_operator_master
+ oc create -f -
catalogsource.operators.coreos.com/service-binding-master created
+ SBO_CATALOG_SOURCE=service-binding-master
+ SBO_SUBSCRIPTION_NAME=service-binding-operator
+ install_redis_operator oc openshift-operators community-operators openshift-marketplace
+ oc create -f -
subscription.operators.coreos.com/my-redis-operator created
+ install_service_binding_operator oc openshift-operators service-binding-operator service-binding-master openshift-marketplace
+ oc create -f -
subscription.operators.coreos.com/my-service-binding-operator created
```
if you see the service binding installation steps catalog source is `service-binding-master` and subscription name is `service-binding-operator` . And the default value set to env are https://github.com/openshift/odo/pull/4958/files#diff-510a35137f161ddca098f30a15bcaac95c9f4f33b75a0bd861a7d8e77a56b1f1R4-R5 . So we change it under Nightly https://github.com/openshift/odo/pull/4958/files#diff-510a35137f161ddca098f30a15bcaac95c9f4f33b75a0bd861a7d8e77a56b1f1R64-R65 .

Reply:
> No its not under if block https://github.com/openshift/odo/pull/4958/files#diff-510a35137f161ddca098f30a15bcaac95c9f4f33b75a0bd861a7d8e77a56b1f1R72

I'm referring only to [this part of the code](https://github.com/openshift/odo/blob/96b8bc6ffc1d6f2e40d3c7746115c4bea467edd2/scripts/configure-cluster/common/setup-operators.sh#L59-L66). That's also what I have commented at. I'm not sure why you're referring to above block. 😕 



> if you see the service binding installation steps catalog source is `service-binding-master` and subscription name is `service-binding-operator` . And the default value set to env are https://github.com/openshift/odo/pull/4958/files#diff-510a35137f161ddca098f30a15bcaac95c9f4f33b75a0bd861a7d8e77a56b1f1R4-R5 . So we change it under Nightly https://github.com/openshift/odo/pull/4958/files#diff-510a35137f161ddca098f30a15bcaac95c9f4f33b75a0bd861a7d8e77a56b1f1R64-R65 .

Right. But why are we changing the values after the call to `deploy_service_binding_operator_master` is over? What impact does it have when done this way? Especially when inside the `deploy_service_binding_operator_master` function, we're using the values directly and not through `SBO_CATALOG_SOURCE` or `SBO_SUBSCRIPTION_NAME` variables.